### PR TITLE
Made maxLines property nullable and added assertions

### LINKED ISSUE #809: Text Field not able to expand
Before null-safety, you could make the FormBuilderTextField expand vertically based on the input when setting `maxLines: null`.

Because `maxLines` is now (version 6.0.1) of type `int` and not `int?`, null cannot be assigned anymore. However, I did not find a reason why maxLines could not be nullable.

This also breaks the `expands` property, because for it not to throw an error, both `minLines` and `maxLines` must be null.

### PR DESCRIPTION
**Pull request by S-ecki**
made `maxLines` property nullable
added assertions for proper use of `expands`

resolves #809

### CODE REVIEW

**Comment by danvick** [comment]
Hi @S-ecki,

Kindly note that your changes have brought about CI build failure. Please resolve the issues before we can merge this PR

**Review by danvick — CHANGES_REQUESTED** [review]
Line 303 is still setting a default value for `maxLines`. 

If `maxLines` is to be nullable, then we'd need to remove the default value for the property. Otherwise, it won't be nullable.

**Comment by S-ecki** [comment]
I believe the problem was a false assertion that I inserted.
Setting a `maxLines` default value should be compatible with it being nullable as Flutter´s `TextFormField` has implemented it the same way. 

However, I could be mistaken and await your review!

**Comment by tahesse** [comment]
Hey, what's the status of this PR? 😅

**Comment by S-ecki** [comment]
> Hey, what's the status of this PR? 😅

I commited changes that (as far as I can tell) solve the issue and do pass the tests, so I am waiting on a response.

Perhaps you can check the PR and see if my changes work for you

**Comment by tahesse** [comment]
> I commited changes that (as far as I can tell) solve the issue and do pass the tests, so I am waiting on a response.
> 
> Perhaps you can check the PR and see if my changes work for you

Your changes work for me, that is why I ask for the status of this PR. It seems that there is one failing test due to less (?) code coverage.

**Comment by S-ecki** [comment]
> Your changes work for me, that is why I ask for the status of this PR. It seems that there is one failing test due to less (?) code coverage.

I do not fully understand the code coverage problem, as my PR did not really change it (didnt add/remove tests and just added some null operators)

Perhaps @danvick can help us out

**Pull request merged**
